### PR TITLE
Update WorkLock terminology

### PR DESCRIPTION
**Pull request by mswilkison**
Fix terminology around WorkLock:
* bid -> escrow
* bidder -> participants
* distribution/portion -> allocation
* distribute -> allocate
* token distribution -> network node setup
* remove instances of `tokens`

### CODE REVIEW

**Comment by KPrasch** [comment]
Does this PR have a defined scope (only docs and python for example) - or can we handle *all* the renames with team participation in *this* PR?  (@vzotova will need to handle solidity renames :sweat_smile: )

**Comment by mswilkison** [comment]
Definitely contribute! :)

I'm ambivalent re: a deep renaming or just a shallow renaming of the API.

File: docs/source/architecture/worklock.rst [hunk 1]
```diff
@@ -7,31 +7,37 @@ WorkLock
...
- - Once all bidders have been assigned the minimum amount of NU, each bidder with a `bonus` will receive a portion of the remaining NU, distributed pro rata across all participants, taking into consideration only their bonus ETH amounts.
- - If the resulting NU distributed to a bidder is above the maximum allowed NU to stake, then such a bidder has their bid partially refunded until the corresponding amount of NU is within the allowed limits.
-
-Finally, if WorkLock participants use that stake-locked NU to run a node, the NU will eventually unlock and their escrowed ETH will be returned in full.
-
-
-Hypothetical Bidding Scenarios
+`WorkLock` is a novel, permissionless network node setup mechanism, developed at NuCypher, which requires participants
+to temporarily stake ETH and operate NuCypher network threshold cryptography nodes in order to be allocated NU, the native
+cryptocurrency of the NuCypher network that enables node operation.
```
Comment (cygnusv) [inline docs/source/architecture/worklock.rst:12]:
```suggestion
token of the NuCypher network that enables node operation.
```
Question: what's the preferred term, token or cryptocurrency?

File: docs/source/architecture/worklock.rst [hunk 2]
```diff
@@ -7,31 +7,37 @@ WorkLock
...
+The WorkLock begins with an escrow period, during which anyone seeking to operate a NuCypher network threshold cryptography node can send
+ETH to the WorkLock contract to be temporarily escrowed on-chain.
+At any time during the escrow period, WorkLock participants can cancel their participation to forgo NU and recoup their escrowed ETH immediately.
+Once the escrow period closes, the WorkLock contract does not accept more ETH, but it will still accept
+cancellations during an additional time window.
+At the end of this cancellation period, the allocation period opens and stake-locked NU are allocated to participants in non-transferable stakes designed for
+the limited purpose of running threshold cryptography nodes on the live NuCypher network.
+Stake-locked NU will be allocated at network launch according to the following principles:
+
+ - All of the NU held by WorkLock will be allocate to network nodes in non-transferable stakes designed to enable node operation.
```
Comment (cygnusv) [inline docs/source/architecture/worklock.rst:26]:
```suggestion
 - All of the NU held by WorkLock will be allocated to network nodes in non-transferable stakes designed to enable node operation.
```

File: docs/source/architecture/worklock.rst [hunk 3]
```diff
@@ -7,31 +7,37 @@ WorkLock
...
+ - All ETH escrows must be greater than or equal to the minimum allowed escrow.
+ - For each escrow, the surplus above the minimum allowed escrow is called the `bonus`; all escrows are composed of a `base` escrow (fixed minimum) and a `bonus` escrow (variable amount).
+ - Each participant will receive at least the minimum amount of staked NU needed to operation a network node.
```
Comment (cygnusv) [inline docs/source/architecture/worklock.rst:29]:
```suggestion
 - Each participant will receive at least the minimum amount of staked NU needed to operate a network node.
```

File: docs/source/architecture/worklock.rst [hunk 4]
```diff
@@ -7,31 +7,37 @@ WorkLock
...
+ - Once all participants have been allocated the minimum amount of NU, each participant with a `bonus` will be allocated a portion of the remaining NU,
+   allocated pro rata across all participants, taking into consideration only their bonus ETH amounts.
+ - If the resulting NU allocated to a participant is above the maximum allowed NU needed to operate a network node, then such a participant has their escrow partially refunded until the corresponding amount of NU is within the allowed limits for node operation.
+
+Finally, if WorkLock participants use that stake-locked NU to run a node and provide threshold cryptography services on the live network for a minimum of six months,
+the NU will subsequently unlock and their temporarily escrowed ETH will be returned in full.
+(Participants that fail to successfully run a NuCypher network node for six months of the live network will not receive NU and their ETH will remain escrowed in the
+WorkLock contract.)
```
Comment (cygnusv) [inline docs/source/architecture/worklock.rst:37]:
```suggestion
Participants that fail to successfully run a NuCypher network node for six months of the live network will not receive NU and their ETH will remain escrowed in the
WorkLock contract.
```

File: docs/source/guides/worklock_guide.rst [hunk 5]
```diff
@@ -105,58 +105,58 @@ The following is an example output of the ``status`` command (hypothetical value
...
 
 .. code::
 
-    WorkLock Participant <BIDDER ADDRESS>
+    WorkLock Participant <PARTICIPANT ADDRESS>
     =====================================================
-    Tokens Claimed? ...... No
-    Total Bid ............ 22 ETH
+    NU Claimed? ...... No
+    Total Escrow ............ 22 ETH
```
Comment (derekpierre) [inline docs/source/guides/worklock_guide.rst:142]:
```suggestion
    Total Escrow ......... 22 ETH
```

File: docs/source/guides/worklock_guide.rst [hunk 6]
```diff
@@ -105,58 +105,58 @@ The following is an example output of the ``status`` command (hypothetical value
...
         Base ETH ......... 15 ETH
         Bonus ETH ........ 7 ETH
-    Tokens Allocated ..... 68000 NU
+    NU Allocated ..... 68000 NU
```
Comment (derekpierre) [inline docs/source/guides/worklock_guide.rst:145]:
```suggestion
    NU Allocated ......... 68000 NU
```

File: docs/source/guides/worklock_guide.rst [hunk 7]
```diff
@@ -105,58 +105,58 @@ The following is an example output of the ``status`` command (hypothetical value
...
+The following output is an example of what is included when ``--participant-address`` is used
 
 .. code::
 
-    WorkLock Participant <BIDDER ADDRESS>
+    WorkLock Participant <PARTICIPANT ADDRESS>
     =====================================================
-    Tokens Claimed? ...... No
-    Total Bid ............ 22 ETH
+    NU Claimed? ...... No
```
Comment (derekpierre) [inline docs/source/guides/worklock_guide.rst:141]:
```suggestion
    NU Claimed? .......... No
```

File: docs/source/guides/worklock_guide.rst [hunk 8]
```diff
@@ -105,58 +105,58 @@ The following is an example output of the ``status`` command (hypothetical value
...
-Alternatively, when the allocated tokens have been claimed, the following is an example of the output
+Alternatively, when the NU has been allocated, the following is an example of the output
 
 .. code::
 
-    WorkLock Participant <BIDDER ADDRESS>
+    WorkLock Participant <PARTICIPANT ADDRESS>
     =====================================================
-    Tokens Claimed? ...... Yes
+    NU Claimed? ...... Yes
```
Comment (derekpierre) [inline docs/source/guides/worklock_guide.rst:159]:
```suggestion
    NU Claimed? .......... Yes
```

File: docs/source/guides/worklock_guide.rst [hunk 9]
```diff
@@ -167,66 +167,66 @@ Alternatively, when the allocated tokens have been claimed, the following is an
...
-    - Tokens Allocated
-        Allocation of NU tokens
+        Surplus over minimum escrow
+    - NU Allocated
+        Allocation of NU
     - Locked ETH
         Remaining ETH to be unlocked via completion of work
-    - Tokens Claimed
-        Whether the allocation of NU tokens have been claimed or not
+    - NU Allocated
```
Comment (derekpierre) [inline docs/source/guides/worklock_guide.rst:180]:
```suggestion
    - NU Claimed
```

File: nucypher/cli/commands/worklock.py [hunk 10]
```diff
@@ -76,8 +76,8 @@
 from nucypher.cli.types import DecimalRange, EIP55_CHECKSUM_ADDRESS
 from nucypher.config.constants import NUCYPHER_ENVVAR_PROVIDER_URI
 
-option_bidder_address = click.option('--bidder-address',
-                                     help="Bidder's checksum address.",
+option_bidder_address = click.option('--participant-address',
+                                     help="Participants's checksum address.",
```
Comment (derekpierre) [inline nucypher/cli/commands/worklock.py:80]:
```suggestion
                                     help="Participant's checksum address.",
```

File: docs/source/architecture/worklock.rst [hunk 11]
```diff
@@ -7,31 +7,37 @@ WorkLock
...
+ - Each participant will receive at least the minimum amount of staked NU needed to operate a network node.
+ - Once all participants have been allocated the minimum amount of NU, each participant with a `bonus` will be allocated a portion of the remaining NU,
+   allocated pro rata across all participants, taking into consideration only their bonus ETH amounts.
+ - If the resulting NU allocated to a participant is above the maximum allowed NU needed to operate a network node, then such a participant has their escrow partially refunded until the corresponding amount of NU is within the allowed limits for node operation.
+
+Finally, if WorkLock participants use that stake-locked NU to run a node and provide threshold cryptography services on the live network for a minimum of six months,
+the NU will subsequently unlock and their temporarily escrowed ETH will be returned in full.
+Participants that fail to successfully run a NuCypher network node for six months of the live network will not receive NU and their ETH will remain escrowed in the
```
Comment (derekpierre) [inline docs/source/architecture/worklock.rst:36]: [stake-rewarded] NU or [minted] NU

```suggestion
Participants that fail to successfully run a NuCypher network node for six months of the live network will not receive stake-rewarded NU and their ETH will remain escrowed in the
```

File: nucypher/cli/literature.py [hunk 12]
```diff
@@ -452,105 +452,102 @@
 # Worklock
```
Comment (KPrasch) [inline nucypher/cli/literature.py:452]: :+1: I am glad we isolated this file now :-)

Comment (cygnusv) [reply nucypher/cli/literature.py:452]: I totally agree. I remember I felt some resistance at the moment but now I'm also glad we made the change.

**Pull request merged**